pages/docs/database-connections/role-based-credentials.mdx

```diff
@@ -0,0 +1,70 @@
+---
+description: Role-based data credentials allow you to control what users can access from the same data connection.
+---
+
+import { Callout } from "nextra-theme-docs";
+
+# Role-Based Database Credentials
+
+All Hashboard data connections can be optionally configured to use additional credentials which can be assigned to users on a per-role basis. 
+
+After setting up database-level permissions within the data warehouse you use with Hashboard, these credentials can be added within Hashboard and then assigned to your Hashboard project's roles.
+
+This enables you to control what data Hashboard users can access when viewing models, saved explorations, and dashboards built from the same data connection.
+Role-based credentials can be used to restrict access to certain tables, or even certain rows or columns within a table, based on a user's role.
+
+### Adding credentials
+
+The first step to adding role-based credentials is to create a [database connection](/docs/database-connections).
+
+Note that the credentials specified in the top-level of the database connection form is the *default* credential. 
+That is, unless specified otherwise, all users will connect via the information provided in the database connection form.
+
+<Callout>
+    If a user doesn't have any specified credentials within their roles, they will use the default credential for the data connection.
+    Therefore, it's best practice for the default credential to be the most restrictive credential permissions-wise.
```

Comment:
While this does seem like a safer recommendation, I think the workflow _could_ end up being the opposite in some cases - it might depend on the company's specific use case. For example: if only a few people can't see certain columns in a db, it might be simpler to make the default the less restrictive permission and add individuals who need restrictions on an ad hoc basis.

I'll defer approval on what we do here to @vndrewlee on the product side. Could see going with this, or could also see describing something more nuanced.

Reply:
I definitely think that different scenarios could call for different permissioning structures. Also I forgot to mention public links here.

Maybe I can change this to something that still calls this out but is less opinionated, like 

"Make sure your users are seeing the data you intend them to: if a user doesn't have any specified credentials within their roles, they will use the default credential for the data connection. Default credentials are also used in public links, if they are enabled." 

interested in hearing what people think abt this though! Could be something we touch on in a blog post as well

Reply:
As per our convo about what we're going with here (the way you suggested!) - let's make sure to make this opinion clearly stated now! Perhaps even more opinionated than it is now, such as:

`Hashboard assumes that your default credential is the most restrictive. etc. etc.`

Reply:
Thanks for pointing this out– I made this warning stronger

"If a user doesn't have any specified credentials within their roles, they will use the default credential for the data connection. Default credentials are also used in public links, if they are enabled.

Therefore, Hashboard assumes your default credential is the most restrictive credential permissions-wise. Make sure your default credential doesn't allow access to sensitive data that you would not want accessible to all users in your organization."